### PR TITLE
Fix crash with invalid private @inlinable function

### PR DESCRIPTION
**Pull request by slavapestov**
We had some duplicated logic between getResilienceExpansion() and getFragileFunctionKind(). Clean this up by moving the latter into AST, and re-implementing the former in terms of the latter.

This also fixes a crash in at least one case where these two implementations had previously diverged.

Fixes <rdar://problem/60605117>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/AST/DeclContext.cpp [hunk 1]
```diff
@@ -312,19 +312,31 @@ bool DeclContext::isGenericContext() const {
...
 }
 
-/// Get the most optimal resilience expansion for the body of this function.
-/// If the body is able to be inlined into functions in other resilience
-/// domains, this ensures that only sufficiently-conservative access patterns
-/// are used.
 ResilienceExpansion DeclContext::getResilienceExpansion() const {
+  auto fragileKind = getFragileFunctionKind();
+  switch (fragileKind.kind) {
+    case FragileFunctionKind::Transparent:
```
Comment (CodaFi) [inline lib/AST/DeclContext.cpp:318]: Nit: LLVM-style switch-cases please.

Comment (slavapestov) [reply lib/AST/DeclContext.cpp:318]: Oops, that was an accident.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
```
15:29:21 The following tests FAILED:
15:29:21 	917 - TestFoundation.TestNotificationQueue-test_notificationQueueLifecycle (Failed)
```

@millenomi Any ideas?

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Pull request merged**